modules/tensor_mechanics/doc/content/modules/tensor_mechanics/1d_elastic_waves.md

```diff
@@ -0,0 +1,98 @@
+    \nabla u= ik \, u \, \qquad at \, x=0,
+    \\
+    u= A \, cos(2 \pi f t) \, \qquad at \, x=L,\label{eq5}
+\end{equation}
+
+Transforming equation (4) and the associated boundary conditions (5) in time to the frequency domain, we obtain:
+\begin{equation}
+     -\nabla \cdot \hat \sigma- \omega^2  U = 0 \label{eq6}
+\end{equation}
+with the following boundary conditions:
+\begin{equation}
+    \nabla U= ik \, U \, \qquad at \, x=0,
+    \\
+U= 0.5 \, \qquad at \, x=L,\label{eq7}
+\end{equation}
+
+where $\hat \sigma = \nabla U$ with $U(x,\omega)$ being the Fourier transform of $u(x,t)$.
+
+!alert note title=Fourier Transfrom of Boundary Conditions
+Not only the pde is transformed to the frequency domain, but also, we must transform all B.Cs as well. Transfroming $A cos(2\pi f t)$ into the frquency domain will result in an impulse at frequency = $f$ with amplitude of A.
+
+When transform the time dependent variable $u$ to the frequency domain, the time-dependency will be omitted. The result is a nother variable $U$ that is frequency dependent. In general, $U$ is complex-valued function/variable in the form of $U_r+iU_i$, where $U_r$ and $U_i$ are the real and the imaginary part of $U$.
+
+At this stage, we split the complex sytem of equations into two $\textbf{real}$ systems, which yields two real-valued systems that live on the same mesh. The strong form PDEs becomes as follows:
+
+\begin{equation}
+    -\nabla \cdot \hat \sigma_r- \omega^2  U_r = 0
+    \\
+    -\nabla \cdot \hat \sigma_i- \omega^2  U_i = 0 \label{eq8}
+\end{equation}
+with the following boundary conditions:
+\begin{equation}
+   \nabla U_r= -k U_i \quad ,\nabla U_i= k U_r\, \qquad at \, x=0,\\
+   U_r=U_i=0 \, \qquad at \, x=L, \label{eq9}
+\end{equation}
+
+Note that this decomposition is exact and there is no infromation missed while decompisition. Also note that, as a result of the radiation BCs on left, $U_r$ and $U_i$ are coupled, hence the two systems in (8).
+
+The following is how we define the kernels and the BCs. of this problem,
+
+!listing examples/wave_propagation/1D_elastic_wave_propagation.i block=Kernels id=listing1
+
+ and here is how we define the BCs. of this problem,
+
+!listing examples/wave_propagation/1D_elastic_wave_propagation.i block=BCs id=listing2
+
+!alert note
+$u$ is a function of time and space $(x,t)$, whereas $U$ is function of space and frequency $(x,\omega)$.
+
+!media media/tensor_mechanics/wavefieldandprofile.png style=width:40%; caption=the solution at $\omega =10$. id=solution
+
+The analytical solution of equation (4) can be straight forward obtained knowing that we can express the solution of $U$ as a combination of sines and cosines representing the traveling and reflecting waves, i.e $U=C_1 e^{ikx} + C_2e^{-ikx}$.
+
+Applying the boundary conditions, and consider that the wave is traveling in the negative $x$ direction, we obtain, $C_1=0$ with $C_2=0.5/(e^{-ikL})$. So, the analytical solution is: $U=0.5 e^{ik(L-x)}.$
```

Comment:
Applying the boundary conditions, and considering the wave to travel in the negative ...

Reply:
done